src/main/java/cuchaz/enigma/Enigma.java

```diff
@@ -56,7 +55,18 @@ public EnigmaProject openJar(Path path, ProgressListener progress) throws IOExce
 
 		services.get(JarIndexerService.TYPE).forEach(indexer -> indexer.acceptJar(classCache, jarIndex));
 
-		return new EnigmaProject(this, classCache, jarIndex);
+		MessageDigest digest;
```

Comment:
I would move this out into a util class tbh, might be useful in other places down the line.